jit/jit-block.c

```diff
@@ -855,6 +916,32 @@ _jit_block_clean_cfg(jit_function_t func)
 				block->ends_in_dead = 1;
 				delete_edge(func, block->succs[1]);
 			}
+			else if(block->num_succs == 2
+				&& is_empty_block(block->next)
+				&& block->next->num_succs == 1
+				&& block->next->succs[0]->flags == _JIT_EDGE_BRANCH
+				&& block->succs[0]->dst == block->next->next)
+			{
+				/* We have a conditional branch, that
+				   branches around the next block, and
+				   the next block consists of just a
+				   jump, like:
+
+					if l7 != 3 then goto .L0
+					goto .L1
+					.L0:
+
```

Comment:
It looks like that in this situation there might be a label on the unconditional branch that is used from elsewhere as a branch target:

    if l7 != 3 then goto .L0
    .L2:
    goto .L1
    .L0:

Hence as far as I understand your proposed transformation is not safe. A safe transformation would look like this:

    if l7 == 3 then goto .L1
    goto .L0
    .L2:
    goto .L1
    .L0:

If .L2 is not actually used then it should probably be optimized away on a next step of the CFG clean algorithm and we will get the optimal and correct result. But if this does not happen then we might get a pessimization instead of optimization.